feat: add hash property

Aman-zishan opened the pull request:
- changes made to importer and exporter contracts

<!--
  😀 Wonderful!  Thank you for opening a pull request.

  Please fill in the information below to expedite the review
  and (hopefully) merge of your change.
-->

### Description of change

<!--
  Please be clear and concise what the change is intended to do,
  why this change is needed, and how you've verified that it
  corrects what you intended.

  In some cases it may be helpful to include the current behavior
  and the new behavior.

  If the change is related to an open issue, you can link it here.
  If you include `Fixes #0000` (replacing `0000` with the issue number)
  when this is merged it will automatically mark the issue as fixed and
  close it.
-->

### Pull-Request Checklist

<!--
  Please make sure to review and check all of the following.

  If an item is not applicable, you can add "N/A" to the end.
-->

- [x] Code is up-to-date with the `main` branch
- [ ] `yarn lint` passes with this change
- [ ] `yarn clarinet:check` passes with this change
- [ ] `yarn clarinet:tests` passes with this 
- [ ] `yarn unit-tests` passes with this
- [ ] Run `yarn eol`
- [ ] Run `yarn format`
- [ ] Run `yarn cleanup`
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

<!--
  🎉 Thank you for contributing!
-->

dorucioclea commented on packages/clarity/contracts/taral/exporter-storage.clar:
```diff
@@ -1,18 +1,30 @@
+(define-map exporter-profile 
+    {
+        exporter-id: uint,
+        hash: (buff 256)
```
Nah, the hash needs to be added in the value side of the map. Two different hashes of the same exporter just mean that the exporter  has changed, but it is still the same exporter. In your case, the `exporter-profile` map could consider the same exporter that has some fields changed as a new exporter, hence, the hash property does not need to be placed in the `key` part of the exporter profile map

dorucioclea reviewed (changes requested):
As described, the hash should not be part of the key, this triggers other changes, probably where the hash is set/get in the set/get functions

dorucioclea commented on packages/clarity/contracts/taral/exporter-storage.clar:
```diff
@@ -62,32 +77,42 @@
     (ok 
         (map-insert orders {id: id,exporter-id: exporter-id}   
-            {order-id: order-id }     
+            {order-id: order-id, hash: hash }     
```
O think particularly for orders, we do not need to keep the hash property here as this is just a map that links exporters to their orders. Apologies for any confusion 😁

Aman-zishan replied:
Makes sense, i'll remove the hash property from here.

dorucioclea commented on packages/clarity/contracts/taral/exporter-storage.clar:
```diff
@@ -62,32 +77,42 @@
 ;; @Desc function to to filter out none from list [none none (some XXX) none]
 ;; @Param value : tuple containing the order ID
-(define-private (is-valid-value (value (optional {order-id: uint})))
+(define-private (is-valid-value (value (optional {order-id: uint, hash: (buff 256)})))
```
As a consequence of my prev comment, hash can be dropped here

dorucioclea commented on packages/clarity/contracts/taral/importer-storage.clar:
```diff
@@ -1,18 +1,32 @@
+    } 
+    {
+        order-id: uint, 
+        hash: (buff 256)
```
Again, we don't need to keep the hash property here

dorucioclea reviewed (changes requested):
Nearly there ✌️

Aman-zishan commented on packages/clarity/contracts/taral/taral-exporter-v1.clar:
```diff
@@ -38,35 +54,48 @@
-            )
+        ;; check that the hash is not empty
+        (asserts! (> (len hash) u0) ERR_EMPTY_HASH)
+        (unwrap! (contract-call? .exporter-storage update-exporter-profile {exporter-id: exporter-id} (merge current-exporter { orders-next-avail-id: (+ u1 new-id)})) exporter-storage-error)
```
@dorucioclea do u think we should merge the hash property here along with incrementing `orders-next-avail-id` cause we are updating the exporter profile and that means change in data right. 

```suggestion
        (unwrap! (contract-call? .exporter-storage update-exporter-profile {exporter-id: exporter-id} (merge current-exporter { orders-next-avail-id: (+ u1 new-id), hash: hash })) exporter-storage-error)
```

Aman-zishan commented:
### CHANGELOG: `taral-purchase-order.clar` -> `taral-purchase-order-v1.clar`

- Variables removed from `order` onchain data
    - delivery country
    - dispatch method
    - shiptment type

- Variables removed from `order-details` onchain data
    - shippping Route list
    - item list

- Removed nested `let` statements and refactored `initialize` function to handle all operations under one `let` statement

### purchase-order-v1 data mappings before and after

<img width="300" alt="Screenshot 2023-07-10 at 7 30 20 PM" src="https://github.com/taraldefi/taral/assets/55238388/55fd8caf-d847-46ad-b132-ed5388bf899f">
<img width="300" alt="Screenshot 2023-07-10 at 7 24 49 PM" src="https://github.com/taraldefi/taral/assets/55238388/c468b8fb-2317-4678-ad4b-9113fba267cf">

Aman-zishan commented:
Closing this as commits replicated in #1301
